### ISSUE TITLE
YOLOv4-Mish Loss very high and negative iou.

### ISSUE THREAD

**Issue by andrelapian** [AlexeyAB/darknet#7643]
this is chart for pre-trained models using yolov4x-mish![mish_olddataset](https://user-images.githubusercontent.com/49072720/115998010-e3214880-a617-11eb-8c2b-8ca9716d608f.png)

This is chart for pre-trained models using yolov4
![6ba354cb-c4a5-46ee-9c6b-48e81d066450](https://user-images.githubusercontent.com/49072720/115998021-e9afc000-a617-11eb-866f-14cbdf15ff0f.png)

My question is why yolov4-mish give high loss even though I use the same dataset?
and why yolov4-mish give negative value for iou_loss?

v3 (iou loss, Normalizer: (iou: 0.05, obj: 4.00, cls: 0.50) Region 168 Avg (IOU: 0.136990), count: 2, class_loss = 39.872097, iou_loss = -29.828011, total_loss = 10.044086 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 1.00, cls: 0.50) Region 185 Avg (IOU: 0.421453), count: 16, class_loss = 5.155934, iou_loss = 0.427526, total_loss = 5.583461 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 0.40, cls: 0.50) Region 202 Avg (IOU: 0.360355), count: 20, class_loss = 1.842278, iou_loss = 2.920083, total_loss = 4.762361 
 total_bbox = 257324, rewritten_bbox = 0.000000 % 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 4.00, cls: 0.50) Region 168 Avg (IOU: 0.000000), count: 1, class_loss = 2.948591, iou_loss = -2.211443, total_loss = 0.737148 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 1.00, cls: 0.50) Region 185 Avg (IOU: 0.449434), count: 17, class_loss = 4.662511, iou_loss = 0.263417, total_loss = 4.925928 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 0.40, cls: 0.50) Region 202 Avg (IOU: 0.467637), count: 27, class_loss = 2.409369, iou_loss = 3.892585, total_loss = 6.301954 
 total_bbox = 257368, rewritten_bbox = 0.000000 % 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 4.00, cls: 0.50) Region 168 Avg (IOU: 0.079922), count: 1, class_loss = 21.372038, iou_loss = -15.975222, total_loss = 5.396816 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 1.00, cls: 0.50) Region 185 Avg (IOU: 0.433072), count: 11, class_loss = 2.610621, iou_loss = 0.336940, total_loss = 2.947561 
v3 (iou loss, Normalizer: (iou: 0.05, obj: 0.40, cls: 0.50) Region 202 Avg (IOU: 0.389636), count: 18, class_loss = 1.677491, iou_loss = 2.662407, total_loss = 4.339898 
 total_bbox = 257398, rewritten_bbox = 0.000000 %

**Comment by molyswu** [comment]
same issues

**Comment by akashAD98** [comment]
same issue. @WongKinYiu @AlexeyAB  your help is really appreciate.
![178loss-weapons](https://user-images.githubusercontent.com/62583018/128819572-bcfa3a36-9478-4e3a-9399-8d28405d00fa.png)
